neurodamus/node.py

```diff
@@ -1421,21 +1472,20 @@ def _sim_corenrn_configure_datadir(self, corenrn_restore, coreneuron_direct_mode
                     (shm_avail >> 20),
                     (mem_avail >> 20),
                 )
-        return corenrn_datadir if not self._shm_enabled else corenrn_datadir_shm
+        SimConfig.coreneuron_datadir = (
+            corenrn_datadir if not self._shm_enabled else corenrn_datadir_shm
+        )
 
     # -
     @timeit(name="corewrite")
-    def _sim_corenrn_write_config(self, corenrn_restore=False):
+    def _coreneuron_write_sim_config(self, corenrn_restore=False):
         log_stage("Dataset generation for CoreNEURON")
-        CoreConfig.datadir = self._sim_corenrn_configure_datadir(
-            corenrn_restore, SimConfig.coreneuron_direct_mode
-        )
 
         if not corenrn_restore:
             CompartmentMapping(self._circuits.global_manager).register_mapping()
             if not SimConfig.coreneuron_direct_mode:
-                with self._coreneuron_ensure_all_ranks_have_gids(CoreConfig.datadir):
-                    self._pc.nrnbbcore_write(CoreConfig.datadir)
+                with self._coreneuron_ensure_all_ranks_have_gids(SimConfig.coreneuron_datadir):
+                    self._pc.nrnbbcore_write(SimConfig.coreneuron_datadir)
```

Comment:
In fact, in `node.py` there are other functions calling `CoreConfig.datadir`.  We need to determine whether to use `SimConfig.coreneuron_datadir` or `CoreConfig.datadir`.
Here `nrnbbcore_write` writes into `SimConfig.coreneuron_datadir`,  but in coreneuron_configuration.py, we link the path in `sim.conf`: `fp.write(f"datpath='{os.path.abspath(self.datadir)}'\n")`. I know they are the same value. But I would prefer to use `CoreConfig.datadir` for `nrnbbcore_write` as well.

Reply:
to me, if it is for reading, it is fine to have a view of the folder in CoreConfig. It is just a matter of convenience as we can also use `SimConfig.coreneuron_input_path()`. If you want I can completely remove `CoreConfig.datadir`. As I said, it was just for convenience

Reply:
I'd like to ensure that `nrnbbcore_write` and coreneuron paramater`--datpath=` points to the same folder. If we use the same variable/obj in these 2 places, it is safe for maintenance. 
Imagine in the future someone changes the CoreConfig.datadir to another folder, and forget `SimConfig.coreneuron_datadir`.  So I prefer the old way `self._pc.nrnbbcore_write(CoreConfig.datadir)`

Reply:
ah I read your change as inverted. That was a typo and thx for the catch! Yes, I am going to change it to `CoreConfig.datadir` ofc. May I close this?